### ISSUE TITLE
Tilesets light occlusion leaks

### ISSUE THREAD

**Issue by jruiss** [godotengine/godot#60532]
### Godot version

v4.0.alpha.custom_build [5c54770b7]

### System information

Windows 10, GTX 1050ti, GLES3

### Issue description

Tileset light occlusion leaks through

![1](https://user-images.githubusercontent.com/63138805/165312769-f56002ca-39f5-43af-860f-a1463dda43cf.PNG)
![2](https://user-images.githubusercontent.com/63138805/165312787-716e1039-ed85-449f-89e4-1b8ba1b9b2f0.PNG)

### Steps to reproduce

1. Create a tileset with a occlusion layer and a tile covered completely by occlusion rectangle
2. Place multiple in a row in a tilemap
3. Create a light with shadow

### Minimal reproduction project

[tilemap_example.zip](https://github.com/godotengine/godot/files/8563674/tilemap_example.zip)

**Comment by Calinou** [comment]
This is likely a duplicate of https://github.com/godotengine/godot/issues/55806.

**Comment by jruiss** [comment]
> Can you reproduce this when using the same Sprite2Ds and LightOccluder2D nodes placed manually? This is likely a duplicate of #55806.

Yeah, the light occluder node got the same problem

**Comment by groud** [comment]
Yeah, I noticed this issue too when working on the TileMap stuff. From my testing, the bug is indeed due to issues in the rendering server internals and thus impacts all LightOccluder2D.
